contract/r/gnoswap/pool/pool_factory_param.gno

```diff
@@ -44,47 +44,48 @@ func newPoolParams(
 	}
 }
 
-func (p *createPoolParams) Token0Path() string {
-	return p.token0Path
-}
-
-func (p *createPoolParams) Token1Path() string {
-	return p.token1Path
-}
-
-func (p *createPoolParams) Fee() uint32 {
-	return p.fee
-}
-
-func (p *createPoolParams) TickSpacing() int32 {
-	return p.tickSpacing
-}
-
-func (p *createPoolParams) SqrtPriceX96() *u256.Uint {
-	return p.sqrtPriceX96
-}
+func (p *createPoolParams) SqrtPriceX96() *u256.Uint { return p.sqrtPriceX96 }
+func (p *createPoolParams) TickSpacing() int32       { return p.tickSpacing }
+func (p *createPoolParams) Token0Path() string       { return p.token0Path }
+func (p *createPoolParams) Token1Path() string       { return p.token1Path }
+func (p *createPoolParams) Fee() uint32              { return p.fee }
 
 func (p *createPoolParams) updateWithWrapping() *createPoolParams {
 	token0Path, token1Path := p.wrap()
 
 	if !p.isInOrder() {
 		token0Path, token1Path = token1Path, token0Path
 
-		oldTick := common.TickMathGetTickAtSqrtRatio(p.sqrtPriceX96)
-		if oldTick > consts.MAX_TICK || oldTick <= consts.MIN_TICK {
+		// Check if initial price is within valid range
+		minSqrtRatio := u256.MustFromDecimal(consts.MIN_SQRT_RATIO)
+		maxSqrtRatio := u256.MustFromDecimal(consts.MAX_SQRT_RATIO)
+
+		if p.sqrtPriceX96.Lt(minSqrtRatio) || p.sqrtPriceX96.Gt(maxSqrtRatio) {
 			panic(newErrorWithDetail(
-				errInvalidTickMinMaxRange,
-				ufmt.Sprintf("The tick value(%d) is out of range due to reverse", oldTick),
+				errOutOfRange,
+				ufmt.Sprintf("sqrtPriceX96(%s) is out of range", p.sqrtPriceX96.ToString()),
 			))
 		}
-		newPrice := common.TickMathGetSqrtRatioAtTick(oldTick * int32(-1))
+
+		// newPrice = 2^192 / oldPrice
+		newPrice := u256.Zero().Div(Q192, p.sqrtPriceX96)
+
 		if newPrice.IsZero() {
```

Comment:
Doesn't it need anymore? because we are checking the range more than least value, `minSqrtRatio`.

Reply:
It couldn't be zero after the range checking.

Reply:
removed https://github.com/gnoswap-labs/gnoswap/pull/612/commits/6cb6ff42cab5721f8181dc4a813de98e25d68d27